File: DI/DI-Dagger-Java/src/main/java/DI/named/AppCollaboratorModule.java
package DI.named;

import dagger.Binds;
import dagger.Module;
import model.named.AppCollaborator;
import model.named.IAppCollaborator;

import javax.inject.Singleton;

@Module
interface AppCollaboratorModule {

    @Singleton
    @Binds
    IAppCollaborator provideAppCollaborator(AppCollaborator impl);
}

File: DI/DI-Dagger-Kotlin/src/main/kotlin/DI/named/AppCollaboratorModule.kt
package DI.named

import dagger.Binds
import dagger.Module
import models.named.AppCollaborator
import models.named.IAppCollaborator
import javax.inject.Singleton

@Module
interface AppCollaboratorModule {

    // Si queremos que sea Singletonm podemos usar @Singleton
    // aquí o en la clase
    // Si lo ponemos en la clase es para todas
    // Si lo ponemos aquí es para este contexto
    @Singleton
    @Binds
    fun provideAppCollaborator(impl: AppCollaborator): IAppCollaborator
}